### PR TITLE
#70: Add support for OFT's command line option `--wanted-tags`

### LINKED ISSUE #70: Add support for CLI option --wanted-tags
See command line option `--wanted-tags [_,]<tag>[,...]`, https://github.com/itsallcode/openfasttrace/blob/main/doc/user_guide.md#import-options

See implementation logic: https://github.com/itsallcode/openfasttrace/blob/main/core/src/main/java/org/itsallcode/openfasttrace/core/cli/commands/AbstractCommand.java#L64-L80

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #70

This also deactivates scheduling for the CodeQL build. GitHub usually deactivates scheduling after some time of inactivity and the build is not monitored anyway.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -5,8 +5,6 @@ on:
     branches: [ main ]
   pull_request:
     branches: [ main ]
-  schedule:
-    - cron: '0 4 * * 3'
```
Comment (tomuben) [inline .github/workflows/codeql-analysis.yml:9]: Any reason why the cron job was removed?
Maybe add a comment in the PR

File: src/test/java/org/itsallcode/openfasttrace/maven/TraceMojoIT.java [hunk 2]
```diff
@@ -209,6 +213,32 @@ void testTracingSelectedArtifactTypes() throws Exception
                 equalTo("ok - 2 total\n"));
     }
 
+    @ParameterizedTest(name = "wanted tags {0} finds {1} items")
+    @CsvSource(delimiter = ';', nullValues = "NULL", value =
+    { "NULL; 3", "tagA; 1", "tagB; 1", "tagA,tagB; 2", "tagA,tagB,_; 3", "tagC; 0" })
```
Comment (tomuben) [inline src/test/java/org/itsallcode/openfasttrace/maven/TraceMojoIT.java:218]: What about:
```suggestion
    { "NULL; 3", "tagA; 1", "tagB; 1", "tagA,tagB; 2", "tagA,tagB,_; 3", "tagC; 0", "_; 1" })
```
a test to filter all entities without tag, right?

Comment (tomuben) [reply src/test/java/org/itsallcode/openfasttrace/maven/TraceMojoIT.java:218]: or maybe also 
```suggestion
    { "NULL; 3", "tagA; 1", "tagB; 1", "tagA,tagB; 2", "tagA,tagB,_; 3", "tagC; 0", "_; 1", "_,tagA; 2" })
```

File: src/main/java/org/itsallcode/openfasttrace/maven/TraceMojo.java [hunk 3]
```diff
@@ -228,14 +257,38 @@ private ImportSettings createImportSettings()
             getLog().info("Tracing doc directory " + docPath.get());
             settings.addInputs(docPath.get());
         }
-        if (artifactTypes != null)
+        final FilterSettings filterSettings = FilterSettings.builder()
+                .artifactTypes(getFilteredArtifactTypes())
+                .tags(getFilteredTags())
+                .withoutTags(isFilterWithoutTags())
```
Comment (tomuben) [inline src/main/java/org/itsallcode/openfasttrace/maven/TraceMojo.java:263]: Ah ok, function `withoutTags()` is already part of OpenFastTrace, found it [here](https://github.com/itsallcode/openfasttrace/blob/159a20ebb89a195f41eed181a8f8ecb44a16521e/api/src/main/java/org/itsallcode/openfasttrace/api/FilterSettings.java#L207)

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (kaklakariada) [reply .github/workflows/codeql-analysis.yml:9]: Added a note.

File: README.md [hunk 4]
```diff
@@ -167,6 +172,24 @@ The following example configuration limits import to artifact types `feat` and `
 </configuration>
 ```
 
+This works similar to OFT's command line argument `--wanted-artifact-types`.
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply README.md:175]: It's similar but not the same ;)

**Pull request merged**